sync/upstream.py

```diff
@@ -944,11 +946,19 @@ def commit_status_changed(git_gecko, git_wpt, sync, context, status, url, sha):
             details = "\n".join(details)
             msg = ("Can't merge web-platform-tests PR due to failing upstream checks:\n%s" %
                    details)
-            with env.bz.bug_ctx(sync.bug) as bug:
-                bug["comment"] = msg
-                commit_author = sync.gecko_commits[0].email
-                if commit_author:
-                    bug.needinfo(commit_author)
+            try:
+                with env.bz.bug_ctx(sync.bug) as bug:
+                    bug["comment"] = msg
+                    commit_author = sync.gecko_commits[0].email
+                    if commit_author:
+                        bug.needinfo(commit_author)
+            except BugsyException:
+                msg = traceback.format_exc()
+                logger.warning("Failed to update bug:\n%s" % msg)
+                # Sometimes needinfos fail because emails addresses in bugzilla don't
+                # match the commits. That's non-fatal, but record the exception here in
+                # case something more unexpected happens
+                newrelic.agent.record_exception()
```

Comment:
would it be worth to record the difference of emails and associate them for future use? Or would that introduce issues?

Reply:
There's probably some way to record the fact that a particular email failed. Then I guess we'd need a manually updated map somewhere to record the bugzilla email address for each user with a different commit email address. It's possible but certainly a followup task.